corehq/apps/userreports/models.py

```diff
@@ -83,6 +83,25 @@ def _check_ids(value):
         raise BadValueError("Invalid ID: '{}'".format(value))
 
 
+class UserReportActionLog(models.Model):
+    BUILD = 0
+    MIGRATE = 1
+    REBUILD = 2
+    DROP = 3
+
+    id = models.BigAutoField(primary_key=True)
+    domain = models.CharField(max_length=126, null=False, db_index=True)
+    indicator_config_id = models.CharField(max_length=126, null=False, db_index=True)
+    date_created = models.DateTimeField(auto_now_add=True, db_index=True)
+    action = models.PositiveSmallIntegerField(default=0, choices=(
+        (BUILD, _('Build')),
+        (MIGRATE, _('Migrate')),
+        (REBUILD, _('Rebuild')),
+        (DROP, _('Drop')),
+    ), db_index=True)
+    context = JSONField()
```

Comment:
What are you thinking about storing here? I think user (or system) who created the action would be useful, but it could be useful to have that as its own column as well.

What about the traceback? The traceback might allow us to keep this creation in one place and have context of the originating code (report builder vs management command vs pillow) not sure how well it would work with celery tasks though

Another useful thing would be automated or manual. that could also be a different column (or tied into action)

Reply:
those are some of the things I was thinking about trying to capture, I'll have a look at the code paths and see what I find.

Reply:
what about something like what we do for case rebuilds: https://github.com/dimagi/commcare-hq/blob/master/corehq/form_processor/models.py#L1504-L1546

Reply:
Seems fine. I'm not sure what tradeoffs we want to consier with storing a json/jsonobject blob versus more relation models or just a wide model with nullable columns.

fwiw you could probably use https://github.com/dimagi/commcare-hq/pull/22690/commits/7ad0bf33e881d2c802dc4ef62076a9e333f494b7 as a basis for a JSONObjectField